ckan/logic/action/update.py

```diff
@@ -290,26 +292,51 @@ def package_update(
     original_package = context.get('original_package')
     copy_resources = {}
     changed_resources = data_dict.get('resources')
+    dataset_changed = True
+
+    if not original_package or original_package.get('id') != pkg.id:
+        show_context = logic.fresh_context(context, ignore_auth=True)
+        original_package = _get_action('package_show')(show_context, {
+            'id': data_dict['id'],
+        })
+
+    if data_dict.get('metadata_modified'):
+        dataset_changed = original_package != data_dict
+    else:
+        dataset_changed = dict(
+            original_package, metadata_modified=None, resources=None
+            ) != dict(data_dict, metadata_modified=None, resources=None)
+    if not dataset_changed and original_package.get('resources'
+            ) == data_dict.get('resources'):
+        # flag no change for caller
+        context['package_update_unchanged'] = 'before validation'
```

Comment:
Not in love with this part but it's the simplest possible way I could think to let the activity extension skip creating useless activity entries. I don't like that this is a new Context value that's a really specific case not a general solution.

Alternatives I considered:
- Create a generic way to return extra information in the API envelope via the context that could be used by other actions. This would be useful for API users so they would know if an update actually occurred after any of the package or resource updating actions.
- Create a new interface for the activity plugin that could be passed both the `original_package` and updated `data_dict` from `package_update` so it doesn't have to call `package_show` itself or and the generic signal handling.

Reply:
We are getting rid of an ugly context var and introducing an ugly context var so it evens out :)

> Create a generic way to return extra information in the API envelope via the context that could be used by other actions. 

You mean something like `context["extra_data"]["package_update_unchanged"] = "before validation"`?
This looks like encouraging sharing info in the context, so perhaps I prefer a specific key that is clear where it comes from. Unless I misunderstood?

> This would be useful for API users so they would know if an update actually occurred after any of the package or resource updating actions.

Not sure what you mean here. You mean an update happening e.g. in a chained action?

Reply:
@amercader I'm thinking something like `context['updated_entities']['package'] = [id]` and then expose that `updated_entities` in the API response envelope or something so that API users can know when something has been modified as a result of the action they call. Wouldn't need to implement it for everything right away but it would naturally extend to resources, groups, orgs, users etc. and bulk actions that may update many things at once.

Reply:
As discussed at meeting I'll implement this as `context['changed_entities']` to potentially encompass creation and deletion in the future, add this value to the response envelope for API calls and document it in the changelog.